src/mxarray.jl

```diff
@@ -655,10 +655,10 @@ function jstring(mx::MxArray)
         throw(ArgumentError("jstring only applies to strings (i.e. char vectors)."))
     end
     len = ncols(mx) + 2
-    tmp = Array(UInt8, len)
+    tmp = Array(UInt8, len-2)
     ccall(_mx_get_string, Cint, (Ptr{Void}, Ptr{UInt8}, mwSize), 
         mx.ptr, tmp, len)
-    bytestring(pointer(tmp))
+    String(tmp)
```

Comment:
I don't think it's a good idea to rely on the hidden null byte, since someday it might not be there and the chance that anyone realizes this code needs to be fixed is low, and I doubt `pop!` has measurable overhead here.